docs/block-processing/overall.md

```diff
@@ -0,0 +1,107 @@
+Given GState `S` and a block `B` 
+return either an error describing why `B` is invalid or new GState `S'`
+  ```
+  verifyAndApplyGState :: GState -> Block -> Either BlockVerificationError GState
+  -- ^ Note, that function definition and types are different in code and are put here for reader's convenience
+  ```
+
+Note, that in theory verification can be stateless (except genesis
+state `S₀`) and be described without mentioning any additional state
+(apart from blocks themselves). But in practice it would be very
+inefficient and inconvenient.
+
+For sake of simplicity, we describe state transition function in two parts:
+* Verification: given GState `S` and a block `B`, check whether `B` is valid (and can be applied to `S`)
+* Modification: given GState `S` and a block `B` (successfully verified against `S`), produce `S'`
+
+### Block payload
+
+Block payload consists of:
+* `UpdatePayload`. 
+  It's checked by update system component and is described in the [Update system consensus rules](us.md).
+* `TxPayload`. 
+  It's checked by transaction processing component and is described in the [Transaction processing](txp.md).
+* TODO: others 
+
+## Unknown data handling
+
+Many types in Cardano-SL are designed in an extensible way. They can
+be extended via softfork in future (e. g. new data can be attached or
+semantics can be changed). Examples are:
+* `TxIn` has two constructors `TxInUtxo` and `TxInUnknown`. The former
+  is just a reference to an unspent output, it's the only `TxIn` type
+  we use currently. `TxInUnknown` contains arbitrary bytestring as
+  payload. Later we may introduce another type of transaction inputs
+  (e. g. to redeem fees). Then old software will parse these inputs
+  as `TxInUnknown`, but new software will parse them as new type of
+  input (and process them appropriately).
+* `Attributes` is basically a map from 1-byte integers to arbitrary
+  values. Some values can be parsed into known data types, other
+  values are treated as _unparsed fields_. `Attributes` are part of
+  various data types, e. g. `Tx`. In version 0 `Tx` always has empty
+  attributes. But in version 1 we may add more data to `Tx` (put it
+  into `Attributes`). In this case new software will parse this data
+  and old software will treat it as unparsed fields.
+* Unknown address type. Each valid address in Cardano-SL has a
+  type. It can be one of fixed address types or unknown address type
+  which can't be interpreted by current version of software.
+
+There are few more examples, but these three should demonstrate the
+general idea. If we encounter unknown data (unparsed fields, unknown
+tx input type, unknown address type, etc.) in a block, there are two
+possible behaviours:
+1. Consider such block invalid.
+2. Do as many checks as we can and ignore checks which can't be done
+   because data is unknown. This behaviour depends on which type of
+   data we are processing, each particular case is described in more
+   details below.
```

Comment:
No, it's not described in details below anymore. Probably it's better to just drop the part of this sentence after the comma.